CONTRIBUTING.md

```diff
@@ -107,10 +107,65 @@ approved, other developers with commit access may commit it for the author. When
 doing so, it is important to retain correct attribution of the contribution.
 Generally speaking, Git handles attribution automatically.
 
+## Development
+
+### Local setup
+
+Install [pyenv](https://github.com/pyenv/pyenv#installation) and [poetry](https://python-poetry.org/docs/#installation)
+
+Install and set python 3.7.8 with pyenv:
+
+```console
+pyenv install 3.7.8
+pyenv local 3.7.8
+```
+
+To validate that your poetry venv is using the correct python version
```

Comment:
Having said all that above, it's probably worth linking to the Poetry installation doc in case a user isn't familiar with Poetry.

If you'd like an example of a very minimal developer setup guide, this is what we've written up for `tartufo`: https://tartufo.readthedocs.io/en/stable/CONTRIBUTING.html#writing-code

Reply:
poetry doc is linked in the beginning of this section, but yeah i agree with the comment. This current setup is the way our team does, i will keep it but add a note saying pyenv is just a recommended way but not a requirement to work on it